src/components/expressions/expression-lists/singletons-list.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import { List } from 'react-native-paper'
+import { Expression, Package, Reference } from 'wollok-ts/dist/model'
+import { useContext } from '../../../context/ContextProvider'
+import { useProject } from '../../../context/ProjectProvider'
+import { isNamedSingleton } from '../../../utils/wollok-helpers'
+
+type ListSingletonsProps = {
+	packageName: string
+	setReference: (ref: Expression) => void
+}
+export function ListSingletons({
+	packageName,
+	setReference,
+}: ListSingletonsProps) {
+	const { project } = useProject()
+	const {
+		actions: { filterBySearch },
+	} = useContext()
+	const singletons = project
+		.getNodeByFQN<Package>(packageName)
+		.descendants()
+		.filter(isNamedSingleton)
+	return (
+		<>
+			{filterBySearch(singletons).map(({ id, name }) => (
+				<List.Item
+					key={id}
+					title={name}
+					onPress={() => setReference(new Reference({ name: name! }))}
```

Comment:
Por como está tipado ahora CREO que ya no hace falta el `!`
```suggestion
					onPress={() => setReference(new Reference({ name }))}
```